client/src/App.jsx

```diff
@@ -33,10 +38,54 @@ class App extends React.Component {
       })
   }
 
+  computeTotalRating() {
+    let averageRatings = [];
+
+    for(let review of this.state.reviews) {
+      let averageRating = 0;
+      for(let rating in review.ratings) {
+        rating = review.ratings[rating];
+        averageRating += rating;
+      }
+      averageRating = averageRating / 6.0;
+      averageRatings.push(averageRating);
+    }
+
+    let totalRating = averageRatings.reduce((total, current) => {
+      return total + current;
+    }, 0) / this.state.reviews.length;
+
+    return totalRating;
+  }
+
+  computeSubRatings() {
+    let subRatings = {
+      accuracy: 0,
+      communication: 0,
+      cleanliness: 0,
+      location: 0,
+      checkin: 0,
+      value: 0
+    };
+    for(let review of this.state.reviews) {
```

Comment:
Just a suggestion (what you have works and is fine):
Instead of hard coding each rating category, you can set `subRatings` initially to an empty `{}`. Then in line 72 as you iterate through each rating category within `review.ratings`, you can either set the category as a new key in `subRatings` with the value 0 or you can add the rating category value if the key already exists.

Reply:
Good idea - thanks for the tip! I'll change this in the next set of commits.